ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs

```diff
@@ -103,20 +103,21 @@ type LambdaAnalyzer() =
             argDeclType.HasTypeDefinition && (getAbbreviatedEntity argDeclType.TypeDefinition).IsDelegate
         | _ -> false
 
-    let isExpressionApplicable (expr: IFSharpExpression) =
+    let isApplicable (expr: IFSharpExpression) (pats: TreeNodeCollection<IFSharpPattern>) =
         match expr with
         | :? IPrefixAppExpr
         | :? IReferenceExpr
         | :? ITupleExpr
-        | :? IUnitExpr -> true
+        | :? IUnitExpr ->
+            if pats.Count = 1 && pats.First().IgnoreInnerParens() :? IUnitPat then false else true
```

Comment:
`then false else true`?

Reply:
We urgently need an analyzer to simplify this kind of code!